### PR TITLE
DOC: Added example to MultiIndex doc string

### PR DESCRIPTION
**Pull request by topper-123**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I think ``MultiIndex`` doc string should have a quick note on how ``MultiIndex`` is typically instantiated, as you normally wouldn't instantiate it directly.

I've added an example to the doc string, as in my experience  ``MultiIndex`` is one of the more complex areas of Pandas, so a quick example is justified. This may be debatable, so if others disagree on having an example here, I can remove the example and just retain the links to the ``.from_*`` methods.

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -68,6 +68,23 @@ class MultiIndex(Index):
...
+
+    >>> arrays = [[1, 1, 2, 2], ['red', 'blue', 'red', 'blue']]
+    >>> mi = MultiIndex.from_arrays(arrays, names=('number', 'color'))
+    >>> mi
+
+    Such a ``MultiIndex`` instance can be used as an index in a Series or
+    DataFrame, e.g.:
+
+    >>> pd.Series([1, 2, 3, 4], index=mi)
+
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: actually adding a See Also section would be good (as then links are generated)

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -68,6 +68,23 @@ class MultiIndex(Index):
...
+    ---------
+    A new ``MultiIndex`` is typically constructed using one of the helper
+    methods :attr:`MultiIndex.from_arrays``, :attr:`MultiIndex.from_tuples``
+    and :attr:`MultiIndex.from_product``. For example (using ``.from_arrays``):
+
+    >>> arrays = [[1, 1, 2, 2], ['red', 'blue', 'red', 'blue']]
+    >>> mi = MultiIndex.from_arrays(arrays, names=('number', 'color'))
+    >>> mi
+
+    Such a ``MultiIndex`` instance can be used as an index in a Series or
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: this last part is not really relevant

you can certainly link to a MI (and Index) from Series / DataFrame constructor though

**Comment by topper-123** [comment]
I've made changes. 

In addition, I've also added a ``Notes`` section, with a link to the user guide section on MultiIndex.

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -68,6 +68,32 @@ class MultiIndex(Index):
...
+    MultiIndex(levels=[[1, 2], ['blue', 'red']],
+           labels=[[0, 0, 1, 1], [1, 0, 1, 0]],
+           names=['number', 'color'])
+
+    See further examples for how to construct a MultiIndex in the doc strings
+    of the mentioned helper methods.
+
+    Notes
+    -----
+    See the :ref:`user guide <advanced>` for more.
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: I think you have to put the full http link here (technically you don't for the built docs), but in a console this won't work.

Comment (topper-123) [reply pandas/core/indexes/multi.py]: Ok. done.

File: pandas/core/indexes/multi.py [hunk 4]
```diff
@@ -68,6 +68,33 @@ class MultiIndex(Index):
...
+
+    See further examples for how to construct a MultiIndex in the doc strings
+    of the mentioned helper methods.
+
+    Notes
+    -----
+    See the `user guide
+    <http://pandas.pydata.org/pandas-docs/stable/advanced.html>`_ for more.
+
+    See also
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: See Also

File: pandas/core/indexes/multi.py [hunk 5]
```diff
@@ -68,6 +68,33 @@ class MultiIndex(Index):
         Copy the meta-data
     verify_integrity : boolean, default True
         Check that the levels/labels are consistent and valid
+
+    Examples
+    ---------
+    A new ``MultiIndex`` is typically constructed using one of the helper
+    methods :attr:`MultiIndex.from_arrays``, :attr:`MultiIndex.from_product``
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: should be :meth:

Comment (topper-123) [reply pandas/core/indexes/multi.py]: Oh, yes.  Fixed.

File: pandas/core/indexes/multi.py (hunk 4 above)
Comment (topper-123) [reply pandas/core/indexes/multi.py]: Ok

**Comment by jorisvandenbossche** [comment]
@topper-123 Thanks! It's really nice to see all those doc improvements!

**Pull request merged**